packages/ui/src/ui/containers/ManageTokens/ManageTokensPasswordModal/ManageTokensPasswordModal.tsx

```diff
@@ -41,26 +42,29 @@ const PasswordModal = (props: PasswordModalProps) => {
                     initialValues={{}}
                     onAdd={async (data) => {
                         setError(undefined);
-
-                        const password = data.getState().values.password;
-
-                        return sha256(password).then((password_sha256) => {
-                            axios
-                                .post(`/api/yt/${ytAuthCluster}/login`, {
-                                    username,
-                                    password,
-                                })
-                                .then(() => props.handleConfirm(password_sha256))
-                                .catch((error) => setError(error?.response?.data || error));
-                        });
+                        const {password, sha256_password} = data.getState().values;
+
+                        const strategy = createPasswordStrategy(
+                            username,
+                            ytAuthCluster,
+                            isSha256Mode(),
+                        );
+
+                        return strategy(sha256_password || password)
+                            .then((hashedPassword) => {
+                                props.handleConfirm(hashedPassword);
+                            })
+                            .catch((error: any) => setError(error?.response?.data || error));
```

Comment:
It should also work if we call `setError(errro)`. Is there a reason to use error.response.data here?